File: frontend/src/types/epic.ts
export const EPIC_STATUS = {
  PLANNING: 'planning',
  ACTIVE: 'active',
  DONE: 'done',
} as const;

export type EpicStatus = (typeof EPIC_STATUS)[keyof typeof EPIC_STATUS];

export interface Epic {
  id: string;
  title: string;
  description: string | null;
  status: EpicStatus;
  created_at: string;
  updated_at: string;
}

export interface CreateEpicRequest {
  title: string;
  description?: string | null;
}

export interface UpdateEpicRequest {
  title?: string;
  description?: string | null;
  status?: EpicStatus;
}

File: workspace/backend/app/models/epic.py
from __future__ import annotations

import enum
from typing import TYPE_CHECKING

from sqlalchemy import String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, TimestampMixin

if TYPE_CHECKING:
    from app.models.story import Story


class EpicStatus(enum.StrEnum):
    PLANNING = "planning"
    ACTIVE = "active"
    DONE = "done"


class Epic(TimestampMixin, Base):
    __tablename__ = "epics"

    title: Mapped[str] = mapped_column(String(200))
    description: Mapped[str | None] = mapped_column(Text, nullable=True, default=None)
    status: Mapped[EpicStatus] = mapped_column(default=EpicStatus.PLANNING)

    stories: Mapped[list[Story]] = relationship(
        back_populates="epic", cascade="all, delete-orphan"
    )
